prrte-2.0.1rc3 require PMIX_DATA_BUFFER_STATIC_INIT

## Background information

Latest prrte v2.0.1rc3 adds use of `PMIX_DATA_BUFFER_STATIC_INIT` that puts prereq of pmix >= 4.1.1rc5.

### What version of the PMIx Reference Server are you using? (e.g., v1.0, v2.1, git master @ hash, etc.)

- prrte 2.0.1rc3

##### What version of PMIx are you using? (e.g., v1.2.5, v2.0.3, v2.1.0, git branch name and hash, etc.)

 - pmix 4.1.1rc5   # should work
 - pmix 4.1.1rc4   # should fail

### Please describe the system on which you are running

* Operating system/version:
* Computer hardware:
* Network type:

-----------------------------

## Details of the problem

Was addition of `PMIX_DATA_BUFFER_STATIC_INIT` in prrte-2.0.1rc3 intentional or by accident?  If intentional, i think this sets the minimum OpenPMIx version to v4.1.1rc5.

Fails during configure/build of PRRTE if not using older version of openpmix.

Do we want to add a configure check or something in docs on min openpmix version?

```
  # Build install openpmix-4.1.1rc4
  # Build prrte-2.0.1rc5
  $ ./configure \
     --prefix=$PWD/_install \
   --enable-prte-prefix-by-default \
   --enable-debug \
   --with-devel-headers \
   --with-libevent=$LIBEVENT_INSTALL_DIR \
   --with-hwloc=$HWLOC_INSTALL_DIR \ 
   --with-pmix=$PMIX_INSTALL_DIR \
 && make 

  ...<snip>...

  CC       base/plm_base_launch_support.lo
base/plm_base_frame.c:54:23: error: ‘PMIX_DATA_BUFFER_STATIC_INIT’ undeclared here (not in a function)
     .tree_spawn_cmd = PMIX_DATA_BUFFER_STATIC_INIT,
                       ^
Makefile:860: recipe for target 'base/plm_base_frame.lo' failed
make[2]: *** [base/plm_base_frame.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/ssd/tjn/projects/pmix/CONFUSED-SCORPION/source/prrte-git-br-2.0.1rc3/src/mca/plm'
Makefile:1806: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/ssd/tjn/projects/pmix/CONFUSED-SCORPION/source/prrte-git-br-2.0.1rc3/src'
Makefile:976: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
```

--- a/NEWS
+++ b/NEWS
@@ -40,6 +40,7 @@ version 1.0.
 +**** NOTE: As of v2.0.1, PRRTE no longer has a dependency on "pandoc"
 +**** for building and installing man pages.
 
+PR #1137: Protect against PMIx v4.1.1-only definitions
 PR #1135: Final update v2.0.1 for rc3
  - Show MCA base params in prte_info
  - Silence variety of warnings

--- a/config/prte_setup_pmix.m4
+++ b/config/prte_setup_pmix.m4
@@ -117,7 +117,7 @@ AC_DEFUN([PRTE_CHECK_PMIX],[
         AS_IF([test "x`ls $pmix_ext_install_dir/include/pmix_version.h 2> /dev/null`" = "x"],
                [AC_MSG_RESULT([not found - assuming pre-v2.0])
                 AC_MSG_WARN([PRTE does not support PMIx versions])
-                AC_MSG_WARN([less than v4.01 as only PMIx-based tools can])
+                AC_MSG_WARN([less than v4.1.0 as only PMIx-based tools can])
                 AC_MSG_WARN([can connect to the server.])
                 AC_MSG_ERROR([Please select a newer version and configure again])],
                [AC_MSG_RESULT([found])
@@ -156,7 +156,7 @@ AC_DEFUN([PRTE_CHECK_PMIX],[
 
         AS_IF([test "$prte_external_pmix_version_found" = "0"],
               [AC_MSG_WARN([PRTE does not support PMIx versions])
-               AC_MSG_WARN([less than v4.1 as only PMIx-based tools can])
+               AC_MSG_WARN([less than v4.1.0 as only PMIx-based tools can])
                AC_MSG_WARN([can connect to the server.])
                AC_MSG_ERROR([Please select a newer version and configure again])])
 

--- a/src/pmix/pmix-internal.h
+++ b/src/pmix/pmix-internal.h
@@ -468,6 +468,17 @@ PRTE_EXPORT int prte_pmix_register_cleanup(char *path, bool directory, bool igno
 #define PMIX_ERROR_LOG(r) \
     prte_output(0, "[%s:%d] PMIx Error: %s", __FILE__, __LINE__, PMIx_Error_string((r)))
 
+#ifndef PMIX_DATA_BUFFER_STATIC_INIT
+    #define PMIX_DATA_BUFFER_STATIC_INIT    \
+    {                                       \
+        .base_ptr = NULL,                   \
+        .pack_ptr = NULL,                   \
+        .unpack_ptr = NULL,                 \
+        .bytes_allocated = 0,               \
+        .bytes_used = 0                     \
+    }
+#endif
+
 END_C_DECLS
 
 #endif

--- a/src/prted/pmix/pmix_server.c
+++ b/src/prted/pmix/pmix_server.c
@@ -676,6 +676,7 @@ int pmix_server_init(void)
         }
     }
 
+#ifdef PMIX_SINGLETON
     /* if we were started to support a singleton, then let the server library know */
     if (NULL != prte_pmix_server_globals.singleton) {
         PMIX_INFO_LIST_ADD(prc, ilist, PMIX_SINGLETON,
@@ -685,6 +686,7 @@ int pmix_server_init(void)
             return rc;
         }
     }
+#endif
 
     /* if we are the MASTER, then we are the scheduler
      * as well as a gateway */

